### DISCUSSION TITLE
Frameworks integration

### DISCUSSION THREAD

**Discussion by Ni55aN** [retejs/rete#635] [Ideas]
As we continue to develop our project, we want to ensure compatibility with a wide range of frameworks through plugin integrations. 

Currently, we have plugins available for **React**, **Vue**, **Angular**, and **Svelte**. However, we want to ensure we are meeting the needs of our community by extending support to other frameworks as well.

Please take a moment to review the comments below, where we have listed the names of frameworks that may have integration possibilities. Feel free to vote for your preferred framework by upvoting, leaving a reaction, or responding to the respective comment.

Additionally, if you believe we have missed any frameworks, please leave a comment with the framework name and a link to repository.

**Comment by Ni55aN** [comment]
[Preact](https://github.com/preactjs/preact)

**Comment by Ni55aN** [comment]
[Solid.js](https://github.com/solidjs/solid)

**Reply by jfspencer** [reply]
@Ni55aN I am starting up a project with ReteJS, and would like a direct solidJS integration. what would it take to create solidJS plugin? what would the cost be to commission you to do the work?

**Reply by Ni55aN** [reply]
Hello @jfspencer 
Unfortunately, I do not have the necessary resources to undertake such work within any deadlines. Considering the number of votes, this integration will be prioritized

**Comment by Ni55aN** [comment]
[Van.js](https://github.com/vanjs-org/van)

**Comment by Ni55aN** [comment]
[Ember.js](https://github.com/emberjs/ember.js)

**Comment by Ni55aN** [comment]
[Cycle.js](https://github.com/cyclejs/cyclejs)

**Comment by Ni55aN** [comment]
[Alpine](https://github.com/alpinejs/alpine)

**Comment by Ni55aN** [comment]
[Backbone.js](https://github.com/jashkenas/backbone)

**Comment by Ni55aN** [comment]
[Hyperapp](https://github.com/jorgebucaran/hyperapp)

**Comment by Ni55aN** [comment]
[Aurelia](https://github.com/aurelia/framework)

**Comment by Ni55aN** [comment]
[Riot](https://github.com/riot/riot)

**Comment by Ni55aN** [comment]
[Mithril](https://github.com/MithrilJS/mithril.js)

**Comment by Ni55aN** [comment]
[Marko](https://github.com/marko-js/marko)

**Comment by Ni55aN** [comment]
[Stimulus](https://github.com/hotwired/stimulus)

**Comment by Immortalin** [comment]
[Qwik](https://github.com/BuilderIO/qwik)

**Comment by SamuelTallet** [comment]
[Lit](https://github.com/lit/lit/)

**Reply by Ni55aN** [reply]
Hard to believe it’s been nearly a year already. Where does the time go? 😅

Btw, here's an update on your request https://x.com/rete_js/status/1826320486667563245

**Reply by SamuelTallet** [reply]
Yes, the time is so flying... Thanks for the Lit integration in Rete 😉 Appreciated!

**Comment by PaulCombal** [comment]
[JavaScript Web Components?](https://developer.mozilla.org/en-US/docs/Web/API/Web_components)

**Reply by Ni55aN** [reply]
Following up on the Web Components update, I want to confirm that the request has been addressed through integration with Lit. I believe this approach will be acceptable

https://x.com/rete_js/status/1826320486667563245

**Comment by mmoerman** [comment]
FYI I created a SolidJS rete plugin. Its still early days, but if you want to try it out, the repo can be found at: 
https://github.com/mmoerman/solid-rete-plugin

and you can npm the package with:

npm i solid-rete-lugin

I create a s smal SolidJS example App with some examples inspired by @Ni55aN example site, the repo for that is:
https://github.com/mmoerman/solid-rete-plugin-examples

I am niether a SolidJS nor a Rete guru, so there's probably still a few things I'll need to sort out before this is really stable.
